ui/test/nuxeo-error.test.js

```diff
@@ -0,0 +1,136 @@
+/**
+@license
+(C) Copyright Nuxeo Corp. (http://nuxeo.com/)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import { fixture, flush, html, isElementVisible } from '@nuxeo/nuxeo-elements/test/test-helpers.js';
+import '../nuxeo-error.js';
+/* eslint-disable no-unused-expressions */
+
+suite('<nuxeo-error>', () => {
+  test('Should display error messages when properties are set', async () => {
+    const nuxeoError = await fixture(
+      html`
+        <nuxeo-error code="404" message="Failed to find layout" url="notfound.html"></nuxeo-error>
+      `,
+    );
+    await flush();
+
+    expect(isElementVisible(nuxeoError)).to.be.true;
+
+    const code = nuxeoError.shadowRoot.querySelector('.code');
+    expect(isElementVisible(code)).to.be.true;
+    expect(code.innerText).to.equal('404');
+
+    const description = nuxeoError.shadowRoot.querySelector('.description');
+    expect(isElementVisible(description)).to.be.true;
+    expect(description.innerText).to.equal('ERROR.404');
+
+    const url = nuxeoError.shadowRoot.querySelector('.url');
+    expect(isElementVisible(url)).to.be.true;
+    expect(url.innerText).to.equal('notfound.html');
+
+    const message = nuxeoError.shadowRoot.querySelector('.message');
+    expect(isElementVisible(message)).to.be.true;
+    expect(message.innerText).to.equal('Failed to find layout');
+  });
+
+  test('Should not display error messages when properties are not set', async () => {
+    const nuxeoError = await fixture(
+      html`
+        <nuxeo-error code="" message="" url=""></nuxeo-error>
```

Comment:
It is a very valid test but it does not match the description, i.e., the properties are set but to empty value.
In JS and how we have the code, the result is the same, but semantically, it is different.